### PR TITLE
feat: add Reloader 'auto' and 'custom' options

### PR DESCRIPTION
**Pull request by hugo-calado**
Modify values.yaml and _helpers.tpl to add Reloader options of "auto" and "custom" to restart deployment upon configmap/secret change.

### CODE REVIEW

**Comment by hugo-calado** [comment]
Working now in validating directly on k8s cluster.

**Comment by hugo-calado** [comment]
Finished validation of this PR.

Created specific namespace, then created generic configmap:
image

Installed Reloader and helm application (autorestart):
image

Verified deployment has correct annotation:
image

Validated that, altering the value of the configmap triggers the creation of new pod with update value:
image

File: application/values.yaml [hunk 1]
```diff
@@ -46,6 +46,14 @@ envVars:
     #- secretRef:
     #    name: env-secrets
 
+#enable automatic restart upon confimgap/secret changes
+reloader:
```
Comment (jfmrm) [inline application/values.yaml:50]: add `enabled` option default to false.

File: application/values.yaml [hunk 2]
```diff
@@ -46,6 +46,14 @@ envVars:
     #- secretRef:
     #    name: env-secrets
 
+#enable automatic restart upon confimgap/secret changes
+reloader: 
+  auto: false
```
Comment (jfmrm) [inline application/values.yaml:51]: default this to true, so if it's enabled it will add auto reload by default.

File: application/values.yaml [hunk 3]
```diff
@@ -46,6 +46,14 @@ envVars:
...
+  custom:
+    #values:
```
Comment (jfmrm) [inline application/values.yaml:53]: can we remove the values node? seems kinda useless semantically.

**Review by jfmrm — CHANGES_REQUESTED** [review]
looks good, liked the approach on the annotations in the helpers.

**Comment by hugo-calado** [comment]
Validation after new changes applied.

## AUTO

Confirmed annotation is being placed correctly on deployment:
image


Validated after changing configmap the deployment gets automatically restarted:
image

The same behaviour happens when using secrets:
image

## CUSTOM
Validated annotations got placed correctly:
image

Changes on secret 'mysecret' trigger a restart:
image

As well as changes on the configmap 'log-config':
image

**Review by jfmrm — APPROVED** [review]

**Pull request merged**